packages/bruno-app/src/components/ResponsePane/Timeline/index.js

```diff
@@ -53,6 +58,22 @@ const Timeline = ({ request, response }) => {
           );
         })}
       </div>
+
+      {response.error ? (
+        <div className="mt-4">
+          <pre className="line error text-red-500 font-bold">{response.error}</pre>
+        </div>
+      ) : null}
+
+      {responseData ? (
+        <div className="mt-4">
+          <pre className="line response font-bold">
+            <span className="arrow">{'<'}</span> DATA
+          </pre>
+
+          <pre className="line response">{responseData}</pre>
```

Comment:
I don't think showing the complete response _here_ is a good idea, without any limitations. Showing bigger responses will cause some massive lag.

Dismissing the error in the response tab would probably be a better idea.

Reply:
Should responses certain content types be ignored say images/videos/whatever (allow user to override from some config file?)

Also, allow user to set a limit on the maximum number of bytes that should be processed, if breached, response wont be processed?

May be better options?

Reply:
Having a look at https://www.iana.org/assignments/media-types/media-types.xhtml#application there is no easy way of finding out if the content type is binary or not therefore we would need to support user configurable debuggable payloads (along side to a size limit).
Having a look at this solution

Reply:
@thekalinga I implemented the discussed idea (PR description is updated see screenshot)
Three new settings have been added to the *general* preferences tab :
 
- Display the response in the timeline tab *(Yes/No)*
- Limit the display to responses which size is lower than the provided threshold (if the user specifies **0** then the setting is ignored and all sizes are considered)
- Limit the display to responses of the corresponding content types, if empty or **'*'** all the content types will be displayed

Reply:
@fredjeck Thank you. I am not a maintainer on this project. Please note that my understanding of this codebase is extremely limited. However, I will share my comments in few days.

In the interim, here are few comments.

For safety reasons, when user specifies nothing, I tend to be more restrictive rather than less restrictive so as to not run out of memory i.e when no size limit is specified, I would want to restrict to a reasonable size but not unlimited. When they specify no content types, we would ideally want to restrict to a specific known types & if user specifies, we can use those.

Also, I see that the complete response is converted to string before being parsed. We might want to verify content type header to get the number of bytes (or) read raw bytes 1st till we reach the threshold before dropping off the buffer on floor if it overflows user specifies size. Currently we seem to read everything into a simple buffer & also assuming it to be interpreted as a string irrespective of the content-type header sent by server.